server/coordinator/procedure/operation/split/split.go

```diff
@@ -71,33 +70,65 @@ type Procedure struct {
 	state procedure.State
 }
 
-func NewProcedure(id uint64, dispatch eventdispatch.Dispatch, storage procedure.Storage, c *cluster.Cluster, schemaName string, shardID storage.ShardID, newShardID storage.ShardID, tableNames []string, targetNodeName string) *Procedure {
+func NewProcedure(id uint64, dispatch eventdispatch.Dispatch, s procedure.Storage, clusterMetadata *metadata.ClusterMetadata, snapShot metadata.Snapshot, schemaName string, shardID storage.ShardID, newShardID storage.ShardID, tableNames []string, targetNodeName string) (procedure.Procedure, error) {
```

Comment:
The input parameters are too many, how about defining a struct called `ProcParams`?